Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -96,56 +95,42 @@ void HeliumAnalyserEfficiency::init() {
   setPropertyGroup(PropertyNames::OUTPUT_FIT_PARAMS, PropertyNames::GROUP_OUTPUTS);
 }
 
+void validateInputWorkspace(MatrixWorkspace_sptr const &workspace, std::map<std::string, std::string> &errorList) {
+  Kernel::Unit_const_sptr unit = workspace->getAxis(0)->unit();
+  if (unit->unitID() != "Wavelength") {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "All input workspaces must be in units of Wavelength.";
+    return;
+  }
+}
+
 /**
  * Tests that the inputs are all valid
  * @return A map containing the incorrect workspace
  * properties and an error message
  */
 std::map<std::string, std::string> HeliumAnalyserEfficiency::validateInputs() {
   std::map<std::string, std::string> errorList;
-  const std::string inputWorkspaceName = getProperty(PropertyNames::INPUT_WORKSPACE);
-  if (!AnalysisDataService::Instance().doesExist(inputWorkspaceName)) {
+  const WorkspaceGroup_sptr wsGroup = getProperty(PropertyNames::INPUT_WORKSPACE);
+
+  if (wsGroup == nullptr) {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
+  } else if (wsGroup->size() != 4) {
     errorList[PropertyNames::INPUT_WORKSPACE] =
-        "The input workspace " + inputWorkspaceName + " does not exist in the ADS.";
+        "The input group workspace must have four periods corresponding to the four spin configurations.";
   } else {
-    const auto ws = AnalysisDataService::Instance().retrieve(getProperty(PropertyNames::INPUT_WORKSPACE));
-    if (!ws->isGroup()) {
-      errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
-    } else {
-      const auto wsGroup = std::dynamic_pointer_cast<WorkspaceGroup>(ws);
-      if (wsGroup->size() != 4) {
-        errorList[PropertyNames::INPUT_WORKSPACE] =
-            "The input group workspace must have four periods corresponding to the four spin configurations.";
-      }
+    for (size_t i = 0; i < wsGroup->size(); ++i) {
+      MatrixWorkspace_sptr const stateWs = std::dynamic_pointer_cast<MatrixWorkspace>(wsGroup->getItem(i));
```

Comment:
This is very superficial, but as there are other changes to make on this PR, could we move the use of `const` here to the left of the variable declaration ("west const"), to be consistent with the rest of the file? So this would become: `const MatrixWorkspace_sptr stateWs = std::dynamic_pointer_cast<MatrixWorkspace>(wsGroup->getItem(i));`